teuthology/suite/run.py

```diff
@@ -147,7 +149,7 @@ def choose_kernel(self):
             log.info("kernel sha1: {hash}".format(hash=kernel_hash))
             kernel_dict = dict(kernel=dict(kdb=True, sha1=kernel_hash))
             if kernel_hash != 'distro':
-                kernel_dict['kernel']['flavor'] = self.args.kernel_flavor
+                kernel_dict['kernel']['flavor'] = self.flavor
```

Comment:
Same as above.  In the future, the notion of kernel flavor should probably be ripped out completely.
```suggestion
                kernel_dict['kernel']['flavor'] = 'basic'
```